### PR TITLE
Install Aquasec scannercli on jenkins base image

### PR DESCRIPTION
**Pull request by segator**
Aquasec is installed on base image if ```JENKINS_AGENT_BASE_AQUA_SCANNERCLI_URL```  it's defined on ods-configuration

Example URL
```
#JENKINS_AGENT_BASE_AQUA_SCANNERCLI_URL=https://<USER>:<PASSWORD>@download.aquasec.com/scanner/6.0.0/scannercli
```

### CODE REVIEW

File: configuration-sample/ods-core.env.sample [hunk 1]
```diff
@@ -205,6 +205,15 @@ JENKINS_AGENT_DOCKERFILE_PATH=Dockerfile.rhel7
 # Latest tested version is v1.217.3.
 JENKINS_AGENT_BASE_SNYK_DISTRIBUTION_URL=https://github.com/snyk/snyk/releases/download/v1.217.3/snyk-linux
 
+#Aqua CLI binary distribution url
```
Comment (michaelsauter) [inline configuration-sample/ods-core.env.sample:208]:
```suggestion
# Aqua CLI binary distribution url
```

File: configuration-sample/ods-core.env.sample [hunk 2]
```diff
@@ -205,6 +205,15 @@ JENKINS_AGENT_DOCKERFILE_PATH=Dockerfile.rhel7
...
+# Leave empty to avoid installing Aqua.
+# Releases are published at https://download.aquasec.com/scanner
+# To Download the aqua scanner cli requires a valid account on aquasec.com
+# Latest tested version is 6.0.0
+#JENKINS_AGENT_BASE_AQUA_SCANNERCLI_URL=https://<USER>:<PASSWORD>@download.aquasec.com/scanner/6.0.0/scannercli
```
Comment (michaelsauter) [inline configuration-sample/ods-core.env.sample:213]:
```suggestion
# Example: https://<USER>:<PASSWORD>@download.aquasec.com/scanner/6.0.0/scannercli
```

File: jenkins/ocp-config/build/bc.yml [hunk 3]
```diff
@@ -175,4 +179,4 @@ objects:
       type: Docker
     successfulBuildsHistoryLimit: 5
     failedBuildsHistoryLimit: 5
-    nodeSelector: null
+    nodeSelector: null
```
Comment (michaelsauter) [inline jenkins/ocp-config/build/bc.yml:182]: Please add EOL in again :)

**Review by michaelsauter — COMMENTED** [review]
Looks great! 👌 

Two questions:

a) tested? ideas how to test this in automation?
b) shouldn't this target master?

**Comment by segator** [comment]
> Looks great! 👌
> 
> Two questions:
> 
> a) tested? ideas how to test this in automation?
> b) shouldn't this target master?

a) tested via cli on pod not yet via shared lib but will be the same ;)
b) yes I will push it to master as well, yo prefer first to master then to 3.x?

**Comment by tbugfinder** [comment]
IMHO, the terms aqua and aquasec should be used consistently.
I'd even replace AQUA_* by AQUASEC_*

File: jenkins/agent-base/Dockerfile.rhel7 [hunk 4]
```diff
@@ -70,6 +71,16 @@ RUN if [ -z $SNYK_DISTRIBUTION_URL ] ; then echo 'Skipping snyk installation!' ;
...
     fi
 
+# Optionally install Aquasec.
+RUN if [ -z $AQUA_SCANNERCLI_URL ] ; then echo 'Skipping AquaSec installation!' ; else echo 'Installing AquaSec... getting binary from' $AQUA_SCANNERCLI_URL \
+    && wget $AQUA_SCANNERCLI_URL -O aquasec \
+    && mv aquasec /usr/local/bin \
+    && chmod +rwx /usr/local/bin/aquasec \
+    && echo 'AquaSec CLI version:' \
+    && aquasec version \
+    && echo 'Aqua installation completed!'; \
```
Comment (tbugfinder) [inline jenkins/agent-base/Dockerfile.rhel7:81]: align with the "Installing...." message, so 'AquaSec installation completed'.

**Comment by metmajer** [comment]
Very cool @tbugfinder @segator 👍

**Comment by michaelsauter** [comment]
I would actually think this should go ONLY in master. It's a bigger change and in order to use it, you'd also need the updated shared library. The Aqua support PR currently targets master there.

**Comment by segator** [comment]
well there are no breaking changes adding this feature neither on the change on the shared lib so I think we could include it super easy to 3.x, if we add it to master(4.x) then it will take longer to have it on prod, what do you think @metmajer ?

File: jenkins/agent-base/Dockerfile.rhel7 (hunk 4 above)
Comment (segator) [reply jenkins/agent-base/Dockerfile.rhel7:81]: Completely agree, I will change it

**Comment by segator** [comment]
All comments resolved, sorry for the delay!! as soon you approve the PR i will create PR for master aswell

**Comment by clemensutschig** [comment]
@segator - is there NO proxy settings needed, or does the CLI grab POSIX settings?

**Comment by segator** [comment]
> @segator - is there NO proxy settings needed, or does the CLI grab POSIX settings?

it uses default linux http_proxy env var

**Comment by michaelsauter** [comment]
👍 but I still think this should go into master and not 3.x :)

**Comment by metmajer** [comment]
@segator Michael is right, this is a feature for ODS 4

**Comment by segator** [comment]
then I close the PR and move it to master ;)

**Comment by segator** [comment]
master merging PR -> https://github.com/opendevstack/ods-core/pull/976/files

**Pull request closed without merging**